src/tests/HDFFile.cxx

```diff
@@ -118,155 +141,323 @@ class T_CommandHandler : public testing::Test {
     return true;
   }
 
+  /// Can supply pre-generated test data for a source on a topic to profile the
+  /// writing.
+  class SourceDataGen {
+  public:
+    string topic;
+    string source;
+    int seed = 0;
+    std::mt19937 rnd;
+    vector<FlatBufs::ev42::fb> fbs;
+    vector<FileWriter::Msg> msgs;
+    // Number of messages already fed into file writer during testing
+    size_t n_fed = 0;
+    /// Generates n test messages which we can later feed from memory into the
+    /// file writer.
+    void pregenerate(int n, int n_events_per_message) {
+      // LOG(4, "generating {} {}...", topic, source);
```

Comment:
Why the commented log here and lines 171, 250? Should they be uncommented or removed?